### PR TITLE
fix prefixing keys

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: etcd_test.go [hunk 1]
```diff
@@ -28,3 +28,24 @@ func Test_Storage(t *testing.T) {
...
+	config.EtcdClient = etcdClient
+	config.Prefix = "test-etcd-prefix"
+
+	storage, err := New(config)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	storagetest.Test(t, storage)
+}
```
Comment (kopiczko) [inline etcd_test.go]: Now I start to love storagetest even more.

File: etcd.go [hunk 2]
```diff
@@ -71,10 +78,9 @@ type Service struct {
 func (s *Service) Create(ctx context.Context, key, value string) error {
 	var err error
 
-	key, err = microstorage.SanitizeKey(key)
-	if err != nil {
-		return microerror.Mask(err)
-	}
+	// The key isn't sanitized in Create, because it uses Put under the
+	// hood. It causes problems when prefix is set, becuase it would be
+	// double added.
```
Comment (xh3b4sd) [inline etcd.go]: "added twice"

File: etcd_test.go [hunk 3]
```diff
@@ -28,3 +28,24 @@ func Test_Storage(t *testing.T) {
...
+		Endpoints: []string{"http://127.0.0.1:2379"},
+	}
+	etcdClient, err := clientv3.New(etcdConfig)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	config := DefaultConfig()
+	config.EtcdClient = etcdClient
+	config.Prefix = "test-etcd-prefix"
```
Comment (xh3b4sd) [inline etcd_test.go]: What happens when the prefix contains slashes? I think it would be good to have a test for this as well.

**Review by xh3b4sd — APPROVED** [review]

File: etcd_test.go (hunk 3 above)
Comment (kopiczko) [reply etcd_test.go]: Prefix is sanitized.

```
+	if config.Prefix != "" {
 +		p, err := microstorage.SanitizeKey(config.Prefix)
 +		if err != nil {
 +			return nil, microerror.Maskf(invalidConfigError, "prefix must be valid: %s", err)
 +		}
 +		config.Prefix = p
 +	}
```

File: etcd_test.go [hunk 4]
```diff
@@ -1,30 +1,61 @@
-// +build integration
```
Comment (kopiczko) [inline etcd_test.go:1]: this file is replaced, look commit by commit
https://github.com/giantswarm/etcdstorage/pull/5/commits

Comment (xh3b4sd) [reply etcd_test.go:1]: Cool. All good.

**Pull request merged**